crates/metadata/src/lib.rs

```diff
@@ -137,6 +139,11 @@ pub struct Source {
     /// with the metadata.
     #[serde(skip_serializing_if = "Option::is_none")]
     pub wasm: Option<SourceWasm>,
+    /// Extra information about the environment in which the contract was built.
+    ///
+    /// Useful for producing deterministic builds.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub build_info: Option<Map<String, Value>>,
```

Comment:
Curious why we convert to loosely typed json `Map` here? Is that for alt. compilers e.g. `solang` to provide arbitrary data here?

Reply:
Yep, other compilers may want to include other info here. It's also useful for us if we decide that we need to add/remove info later on